internal/pkg/aead/aead.go

```diff
@@ -25,17 +25,18 @@ type Cipher interface {
 // Using an AEAD is a cipher providing authenticated encryption with associated data.
 // For a description of the methodology, see https://en.wikipedia.org/wiki/Authenticated_encryption
 type MiscreantCipher struct {
-	aead cipher.AEAD
+	secret []byte
 }
 
 // NewMiscreantCipher returns a new AES Cipher for encrypting values
 func NewMiscreantCipher(secret []byte) (*MiscreantCipher, error) {
```

Comment:
This is a somewhat minor point, but it feels surprising for this type to be named `MiscreantCipher`, if the object in question (the Miscreant cipher) is only instantiated at `Encrypt`/`Decrypt`-time. Maybe it would be better if we keep the `cipher.AEAD` object owned by the `MiscreantCipher` class, and instead initialize a new `MiscreantCipher` during `Marshal`/`Unmarsal`. Might make it easier to reason about the types, in the future.

What do you think?

Reply:
I'm okay with changing the type name, but I think whatever we want to expose, we want to be `go-routine` safe. I would argue that this should include making both `Encrypt` and `Decrypt` go-routine safe which currently necessitates the instantiation of new objects.

Reply:
I think _just_ creating new objects in the `Unmarshal` and `Marshal` such that those are the only two methods that would be go-routine safe isn't sufficient for what we want to expose from this API.

Reply:
FWIW the `MiscreantCipher` is an implementation of the `Cipher` interface (which all might be poorly named) and has `Unmarshal` and `Marshal` as well as `Encrypt` and `Decrypt` functions. This probably could be structured differently but  we'd just need to reflect those changes in the `MockCipher`